lib/OpenQA/Scheduler/Scheduler.pm

```diff
@@ -669,6 +669,44 @@ sub _job_stop_children {
     }
 }
 
+sub _carry_over_labels {
+    my ($job) = @_;
+    if ($job->comments > 0) {
+        log_debug("not carrying over labels from previous jobs as there are already comments on current");
+        return;
+    }
+
+    # search for previous jobs
+    my @scenario_keys  = qw/DISTRI VERSION FLAVOR ARCH TEST/;
+    my %js_settings    = map { $_ => $job->settings_hash->{$_} } @scenario_keys;
+    my $subquery       = schema->resultset("JobSettings")->query_for_settings(\%js_settings);
+    my @conds          = [{'me.state' => 'done'}, {id => {'<', $job->id}}, {'me.id' => {-in => $subquery->get_column('job_id')->as_query}}];
+    my $limit_previous = 200;                                                                                                                  # arbitrary limit not to search all jobs
+    my %attrs          = (
+        rows     => $limit_previous,
+        order_by => ['me.id DESC']);
+    my $previous_jobs = schema->resultset("Jobs")->search({-and => \@conds}, \%attrs);
+
+    # loop through all comments on all previous jobs in same scenario
+    # if comment is label OR bug carry over comment to current_job
+    while (my $prev = $previous_jobs->next) {
+        my $comments = schema->resultset('Comments')->search({job_id => $prev->id}, {order_by => {-desc => 'me.id'}});
+        while (my $comment = $comments->next) {
+            if ($comment->bugref or $comment->label) {
+                my $rs = $job->comments->create(
+                    {
+                        text => $comment->text,
+                        # TODO can we also use another user id to tell that
+                        # this comment was created automatically and not by a
+                        # human user?
+                        user_id => $comment->user_id,
```

Comment:
I am thinking about extending this further down the road to keep a bit of history, e.g. encode the original author but create the new comment by "the machine". So let's stick the the original author for now and extend this later.